### PR TITLE
data service: support OpenAPI Specification

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

data service: support OpenAPI Specification

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@clark1013 PTAL

File: tidb-cloud/data-service-manage-github-connection.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ To import configurations of an existing Data App to a new Data App, take the fol
 2. On the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project, [create a new Data App](/tidb-cloud/data-service-manage-data-app.md#create-a-data-app) without connecting to GitHub.
 3. [Connect your new Data App to GitHub](#step-1-connect-your-data-app-to-github) with **Auto Sync & Deployment** enabled. When you specify the target repository, branch, and directory for your new Data App, use your new path with the copied configuration files.
 4. Get the ID and name of your new Data App. You can click the name of your new Data App in the left pane and get the App ID and name in the **Data App Properties** area of the right pane.
-5. In your new path on GitHub, update the `app_id` and `app_name` in the `datapp_config.json` file to the ID and name you get, and then push the changes.
+5. In your new path on GitHub, update the `app_id` and `app_name` in the `datapp_config.json` file to the ID and name you get, and then push the changes.<!--TODO: Whether to get app_version and description or not.-->
```
Comment (clark1013) [inline tidb-cloud/data-service-manage-github-connection.md:123]: 这里应该不用特别强调 app_version 和 description，非必填项。

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+---
+
+# Use the OpenAPI Specification of a Data App with Next.js
+
+## Before you begin
+
+Before using OpenAPI Specification with Next.js, make sure that you have the following:
+
+- A [TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md)
+<!--TODO: add yarn-->
```
Comment (clark1013) [inline tidb-cloud/data-service-oas-with-nextjs.md:13]: - Install node, [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) and [yarn](https://yarnpkg.com/getting-started/install)

**Review by clark1013 — CHANGES_REQUESTED** [review]

**Review by clark1013 — APPROVED** [review]

File: tidb-cloud/data-service-manage-data-app.md [hunk 3]
```diff
@@ -106,6 +107,49 @@ To manage deployments, perform the following steps:
...
+
+### Download the OpenAPI Specification
+
+To download the OpenAPI Specification in JSON or YAML format, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
+2. In the left pane, click the name of your target Data App to view its details.
+3. In the **OpenAPI Specification** area, click **Download** and select **JSON** or **YAML**.
+
+    If this is your first time downloading the OpenAPI Specification, the **Authorize Request** dialog box is displayed. Click **Authorize** to authorize the request.
```
Comment (clark1013) [inline tidb-cloud/data-service-manage-data-app.md:124]: 这里文案调整了一下 **Authorization Request** dialog box

File: tidb-cloud/data-service-manage-data-app.md [hunk 4]
```diff
@@ -106,6 +107,49 @@ To manage deployments, perform the following steps:
...
+
+Data Service (beta) provides autogenerated OpenAPI documentation for each Data App. In the documentation, you can view the endpoints, parameters, and responses, and try out the endpoints.
+
+To access the OpenAPI documentation, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
+2. In the left pane, click the name of your target Data App to view its details.
+3. In the **OpenAPI Specification** area, click **Docs**.
+
+    If this is your first time using the OpenAPI Specification, the **Authorize Request** dialog box is displayed. Click **Authorize** to authorize the request.
```
Comment (clark1013) [inline tidb-cloud/data-service-manage-data-app.md:138]: 这里也一样 Authorization Request

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+1. Create a `.env.local` file in the project directory and set the values to the public key and private key of your Data App.
+
+    ```
+    TIDBCLOUD_DATA_SERVICE_PUBLIC_KEY=YOUR_PUBLIC_KEY
+    TIDBCLOUD_DATA_SERVICE_PRIVATE_KEY=YOUR_PRIVATE_KEY
+    ```
+
+    To create an API key for a Data App, see [Create an API key](/tidb-cloud/data-service-api-key.md#create-an-api-key).
+
+2. Replace the content of `app/page.tsx` with the following code, which fetches data from the endpoint and renders it on the page:
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:153]:
```suggestion
2. In the `hello-repos` project directory, replace the content of `app/page.tsx` with the following code, which fetches data from the `GET /repositories` endpoint and renders it on the page:
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+            ))}
+          </ul>
+        </main>
+      )
+    }
+    ```
+
+    > **Note:**
+    >
+    > If your Data App is deployed across multiple regions, you need to configure the endpoint path in `config`. Modify the `config` object as shown below:
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:181]:
```suggestion
    > If your Data App is deployed across multiple regions, you also need to configure the endpoint path in the `config` object as follows:
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    >
+    >  ```js
+    >  const config = new Configuration({
+    >      username: process.env.TIDBCLOUD_DATA_SERVICE_PUBLIC_KEY,
+    >      password: process.env.TIDBCLOUD_DATA_SERVICE_PRIVATE_KEY,
+    >      basePath: "https://${YOUR_REGION}.data.dev.tidbcloud.com/api/v1beta/app/${YOUR_DATA_APP_ID}/endpoint"
+    >    });
+    >  ```
+    >
+    > Make sure to replace `basePath` with the actual endpoint path of your Data App.
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:191]:
```suggestion
    > Make sure to replace `basePath` with the actual endpoint path of your Data App. To get  `${YOUR_REGION}` and `{YOUR_DATA_APP_ID}`, check the **Endpoint URL** in the endpoint **Properties** pane.
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+          </ul>
+        </main>
+      )
+    }
+    ```
+
+    > **Note:**
+    >
+    > If your Data App is deployed across multiple regions, you need to configure the endpoint path in `config`. Modify the `config` object as shown below:
+    >
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:182]: 什么情况下算是 Data App is deployed across multiple regions 呢。

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+-- Insert some sample data into the table
+INSERT INTO repository (name, url)
+VALUES ('tidb', 'https://github.com/pingcap/tidb'),
+        ('tikv', 'https://github.com/tikv/tikv'),
+        ('pd', 'https://github.com/tikv/pd'),
+        ('tiflash', 'https://github.com/pingcap/tiflash');
+```
+
+## 2. Create a Data App
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:40]:
```suggestion
## Step 2. Create a Data App
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+After the data is inserted, navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page in the [TiDB Cloud console](https://tidbcloud.com/). Create a Data App with the `GET /repositories` endpoint and link the Data App to your TiDB Serverless cluster. The corresponding SQL query for this endpoint is as follows, which fetches all items from the `test.repository` table:
+
+```sql
+SELECT * FROM test.repository;
+```
+
+For more information, see [Get started with Data Service](/tidb-cloud/data-service-get-started.md)
+
+## 3. Generate client code
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:50]:
```suggestion
## Step 3. Generate client code
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 11]
```diff
@@ -40,17 +41,17 @@ To create a Data App for your project, perform the following steps:
...
 
-## Modify a Data App
-
-You can rename a Data App, and manage its API keys, linked clusters, and endpoints.
+## Configure a Data App
 
-### Rename a Data App
+You can edit the name, version, or description of a Data App, and manage its GitHub connection, linked data sources, API keys, endpoints, and deployments.
+<!--TODO: Add Description-->
+### Edit properties
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:48]:
```suggestion
### Edit Data App properties
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 12]
```diff
@@ -40,17 +41,17 @@ To create a Data App for your project, perform the following steps:
...
 
-To rename a Data App, perform the following steps:
+You can edit the name, version, and description of a Data App. To edit a Data App, perform the following steps:
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:50]:
```suggestion
You can edit the name, version, and description of a Data App. To edit Data App properties, perform the following steps:
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 13]
```diff
@@ -40,17 +41,17 @@ To create a Data App for your project, perform the following steps:
...
+3. In the **Data App Properties** area, click <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" color="gray.1"><path d="M11 3.99998H6.8C5.11984 3.99998 4.27976 3.99998 3.63803 4.32696C3.07354 4.61458 2.6146 5.07353 2.32698 5.63801C2 6.27975 2 7.11983 2 8.79998V17.2C2 18.8801 2 19.7202 2.32698 20.362C2.6146 20.9264 3.07354 21.3854 3.63803 21.673C4.27976 22 5.11984 22 6.8 22H15.2C16.8802 22 17.7202 22 18.362 21.673C18.9265 21.3854 19.3854 20.9264 19.673 20.362C20 19.7202 20 18.8801 20 17.2V13M7.99997 16H9.67452C10.1637 16 10.4083 16 10.6385 15.9447C10.8425 15.8957 11.0376 15.8149 11.2166 15.7053C11.4184 15.5816 11.5914 15.4086 11.9373 15.0627L21.5 5.49998C22.3284 4.67156 22.3284 3.32841 21.5 2.49998C20.6716 1.67156 19.3284 1.67155 18.5 2.49998L8.93723 12.0627C8.59133 12.4086 8.41838 12.5816 8.29469 12.7834C8.18504 12.9624 8.10423 13.1574 8.05523 13.3615C7.99997 13.5917 7.99997 13.8363 7.99997 14.3255V16Z" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path></svg>, modify the **App Name**, **App Version**, or **Description** field, and then click **Confirm**.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:54]:
```suggestion
3. In the **Data App Properties** area, click <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" color="gray.1"><path d="M11 3.99998H6.8C5.11984 3.99998 4.27976 3.99998 3.63803 4.32696C3.07354 4.61458 2.6146 5.07353 2.32698 5.63801C2 6.27975 2 7.11983 2 8.79998V17.2C2 18.8801 2 19.7202 2.32698 20.362C2.6146 20.9264 3.07354 21.3854 3.63803 21.673C4.27976 22 5.11984 22 6.8 22H15.2C16.8802 22 17.7202 22 18.362 21.673C18.9265 21.3854 19.3854 20.9264 19.673 20.362C20 19.7202 20 18.8801 20 17.2V13M7.99997 16H9.67452C10.1637 16 10.4083 16 10.6385 15.9447C10.8425 15.8957 11.0376 15.8149 11.2166 15.7053C11.4184 15.5816 11.5914 15.4086 11.9373 15.0627L21.5 5.49998C22.3284 4.67156 22.3284 3.32841 21.5 2.49998C20.6716 1.67156 19.3284 1.67155 18.5 2.49998L8.93723 12.0627C8.59133 12.4086 8.41838 12.5816 8.29469 12.7834C8.18504 12.9624 8.10423 13.1574 8.05523 13.3615C7.99997 13.5917 7.99997 13.8363 7.99997 14.3255V16Z" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path></svg>, modify the App name, version, or description, and then click **Confirm**.
```

File: tidb-cloud/data-service-manage-data-app.md (hunk 3 above)
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:124]:
```suggestion
    If this is your first time downloading the OpenAPI Specification, you need to authorize the request when prompted.
```

File: tidb-cloud/data-service-manage-data-app.md (hunk 4 above)
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:138]:
```suggestion
    If this is your first time using the OpenAPI Specification, you need to authorize the request when prompted.
```

File: tidb-cloud/data-service-oas-with-nextjs.md (hunk 8 above)
Comment (Oreoxmt) [reply tidb-cloud/data-service-oas-with-nextjs.md:186]: @clark1013 PTAL

File: tidb-cloud/data-service-manage-data-app.md [hunk 14]
```diff
@@ -106,6 +107,49 @@ To manage deployments, perform the following steps:
...
 
+<!--TODO: add OAS-->
+
+## Use the OpenAPI Specification
+
+Data Service (beta) supports generating the OpenAPI Specification 3.0 for each Data App, which enables you to interact with your endpoints in a standardized format. You can use this specification to generate standardized OpenAPI documentation, client SDKs, and server stubs.
+
+### Download the OpenAPI Specification
+
+To download the OpenAPI Specification in JSON or YAML format, perform the following steps:
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:118]:
```suggestion
To download the OpenAPI Specification in JSON or YAML format for a Data App, perform the following steps:
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 15]
```diff
@@ -106,6 +107,49 @@ To manage deployments, perform the following steps:
...
+    If this is your first time using the OpenAPI Specification, you need to authorize the request when prompted.
+
+4. Then, the OpenAPI documentation is opened in a new tab. In the documentation, you can view the following information:
+
+    - Data App name, version, and description.
+    - Endpoints grouped by tags.
+
+5. (Optional) To try out an endpoint, take the following steps:
+
+    1. Click **Authorize** and enter your public key as **Username** and private key as **Password** in the displayed dialog box.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:147]:
```suggestion
    1. Click **Authorize** and enter your Data App public key as **Username** and private key as **Password** in the displayed dialog box.
```

File: tidb-cloud/data-service-manage-endpoint.md [hunk 16]
```diff
@@ -53,10 +53,10 @@ In TiDB Cloud Data Service, you can generate one or multiple endpoints automatic
...
 
-        - If a batch operation is selected, TiDB Cloud Data Service appends `_batch` to the name of the generated endpoint. For example, if the selected table name is `sample-table` and the selected operation is **POST Batch Create**, the generated endpoint is displayed as `POST sample-table_batch`.
-        - If there has been already an endpoint with the same request method and endpoint name, TiDB Cloud Data Service appends `_copy` to the name of the generated endpoint. For example, `sample-table_copy`.
+        - If a batch operation is selected, the endpoint name is in the `-<selected table name>s` format. For example, if the selected table name is `sample-table` and the selected operation is **POST Batch Create**, the generated endpoint is displayed as `POST -sample-tables`.
+        - If there is already an endpoint with the same request method and endpoint name (`-<selected table name>`), TiDB Cloud Data Service appends `_copy` to the name of the generated endpoint. For example, `-sample-table_copy`.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-endpoint.md:59]: Please check if UI has this change.

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 17]
```diff
@@ -0,0 +1,209 @@
...
+
+4. Generate the client code:
+
+    ```shell
+    yarn run openapi-generator-cli generate -i oas/doc.json --generator-name typescript-fetch -o gen/api
+    ```
+
+    This command generates the client code using the `oas/doc.json` specification as input and outputs the client code to the `gen/api` directory.
+
+## Step 4. Develop your application
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:144]:
```suggestion
## Step 4. Develop your Next.js application
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+
+You can use the generated client code to develop your application.
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:146]:
```suggestion
You can use the generated client code to develop your Next.js application.
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+    >  const config = new Configuration({
+    >      username: process.env.TIDBCLOUD_DATA_SERVICE_PUBLIC_KEY,
+    >      password: process.env.TIDBCLOUD_DATA_SERVICE_PRIVATE_KEY,
+    >      basePath: "https://${YOUR_REGION}.data.dev.tidbcloud.com/api/v1beta/app/${YOUR_DATA_APP_ID}/endpoint"
+    >    });
+    >  ```
+    >
+    > Make sure to replace `basePath` with the actual endpoint path of your Data App. To get `${YOUR_REGION}` and `{YOUR_DATA_APP_ID}`, check the **Endpoint URL** in the endpoint **Properties** panel.
+
+## 5. Preview your application
```
Comment (qiancai) [inline tidb-cloud/data-service-oas-with-nextjs.md:197]:
```suggestion
## 5. Preview your Next.js application
```

File: TOC-tidb-cloud.md [hunk 20]
```diff
@@ -214,6 +214,7 @@
   - [Manage Endpoint](/tidb-cloud/data-service-manage-endpoint.md)
   - [API Key](/tidb-cloud/data-service-api-key.md)
   - [Deploy Automatically with GitHub](/tidb-cloud/data-service-manage-github-connection.md)
+  - [Use OpenAPI Specification With Next.js](/tidb-cloud/data-service-oas-with-nextjs.md)
```
Comment (Oreoxmt) [inline TOC-tidb-cloud.md:217]:
```suggestion
  - [Use OpenAPI Specification with Next.js](/tidb-cloud/data-service-oas-with-nextjs.md)
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 21]
```diff
@@ -106,6 +107,47 @@ To manage deployments, perform the following steps:
...
+
+Data Service (beta) supports generating the OpenAPI Specification 3.0 for each Data App, which enables you to interact with your endpoints in a standardized format. You can use this specification to generate standardized OpenAPI documentation, client SDKs, and server stubs.
+
+### Download the OpenAPI Specification
+
+To download the OpenAPI Specification in JSON or YAML format for a Data App, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
+2. In the left pane, click the name of your target Data App to view its details.
+3. In the **OpenAPI Specification** area, click **Download** and select **JSON** or **YAML**.
```
Comment (clark1013) [inline tidb-cloud/data-service-manage-data-app.md:120]: UI 上有改动，这里换成了 API Specification，避免跟 Admin API 歧义

File: tidb-cloud/data-service-manage-data-app.md [hunk 22]
```diff
@@ -106,6 +107,47 @@ To manage deployments, perform the following steps:
...
+
+4. Then, the OpenAPI documentation is opened in a new tab. In the documentation, you can view the following information:
+
+    - Data App name, version, and description.
+    - Endpoints grouped by tags.
+
+5. (Optional) To try out an endpoint, take the following steps:
+
+    1. Click **Authorize** and enter your Data App public key as **Username** and private key as **Password** in the displayed dialog box.
+        For more information, see [Manage an API key](/tidb-cloud/data-service-api-key.md).
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:146]:
```suggestion
    1. Click **Authorize** and enter your Data App public key as **Username** and private key as **Password** in the displayed dialog box.

        For more information, see [Manage an API key](/tidb-cloud/data-service-api-key.md).

```

File: tidb-cloud/data-service-app-config-files.md [hunk 23]
```diff
@@ -79,6 +81,8 @@ The description of each field is as follows:
 | `app_id`   | String | The Data App ID. Do not change this field unless your `dataapp_config.json` file is copied from another Data App and you want to update it to the ID of your current Data App. Otherwise, the deployment triggered by this modification will fail. |
 | `app_name` | String | The Data App name. |
 | `app_type` | String | The Data App type, which can only be `"dataapi"`. |
+| `app_version` | String | The Data App version. |
```
Comment (Oreoxmt) [inline tidb-cloud/data-service-app-config-files.md:84]:
```suggestion
| `app_version` | String | The Data App version, which is in the `"<major>.<minor>.<patch>"` format. For example, `"1.0.0"`. |
```
@qiancai PTAL

Comment (qiancai) [reply tidb-cloud/data-service-app-config-files.md:84]: LGTM

**Review by qiancai — APPROVED** [review]

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 24]
```diff
@@ -0,0 +1,209 @@
...
+            ))}
+          </ul>
+        </main>
+      )
+    }
+    ```
+
+    > **Note:**
+    >
+    > If your Data App is deployed across multiple regions, you also need to configure the endpoint path in the `config` object as follows:
```
Comment (Oreoxmt) [inline tidb-cloud/data-service-oas-with-nextjs.md:185]:
```suggestion
    > If your Data App is deployed across multiple regions, which means it is linked to clusters in different regions, you will see multiple items in the `servers` section of the downloaded OpenAPI Specification file. You also need to configure the endpoint path in the `config` object as follows:
```

File: tidb-cloud/data-service-oas-with-nextjs.md [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+This document introduces how to use the OpenAPI Specification of a [Data App](/tidb-cloud/tidb-cloud-glossary.md#data-app) to generate client code and develop a Next.js application.
+
+## Before you begin
+
+Before using OpenAPI Specification with Next.js, make sure that you have the following:
+
+- A TiDB cluster. For more information, see [Create a TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md) or [Create a TiDB Dedicated cluster](/tidb-cloud/create-tidb-cluster.md).
+- [Node.js](https://nodejs.org/en/download).
+- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+- [yarn](https://yarnpkg.com/getting-started/install).
```
Comment (Oreoxmt) [inline tidb-cloud/data-service-oas-with-nextjs.md:17]:
```suggestion
- [Node.js](https://nodejs.org/en/download)
- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
- [yarn](https://yarnpkg.com/getting-started/install)
```

File: tidb-cloud/data-service-oas-with-nextjs.md (hunk 24 above)
Comment (qiancai) [reply tidb-cloud/data-service-oas-with-nextjs.md:185]: If your Data App is deployed across multiple regions, which means its linked clusters are hosted in different regions, you will see multiple items in the `servers` section of the downloaded OpenAPI Specification file. In this case, you also need to configure the endpoint path in the `config` object as follows:

**Pull request merged**